.github/workflows/cpp.yml

```diff
@@ -55,23 +55,42 @@ jobs:
         addlicense -check .
 
   build:
+    name: build (${{ matrix.runtime }})
+
     runs-on: ubuntu-latest
 
     strategy:
       matrix:
-        # TODO(mathetake): Add other runtimes.
-        runtime: [ "wasmtime" ]
+        # runtime: Wasm runtime.
+        # cache_key_regex: used for calculating Bazel cache key.
+        include:
+          # TODO(mathetake): Add V8.
+          - runtime: wamr
+            cache_key_regex: bytecodealliance/wasm-micro-runtime|llvm-project
+          - runtime: wasmtime
+            cache_key_regex: bytecodealliance/wasmtime
+          - runtime: wavm
+            cache_key_regex: WAVM/WAVM|llvm-project
 
     steps:
     - uses: actions/checkout@v2
 
+    - name: Install dependency
+      run: sudo apt-get install ninja-build
+
+    # Calculate the cache key by greping target repositories in repositories.bzl with "matrix.cache_key_regex".
+    # TODO(mathetake): Use native Bazel solution if possible.
+    - name: Setup Bazel cache key
+      run: | 
+        GHA_BAZEL_CACHE_KEY=$(grep -E "${{ matrix.cache_key_regex }}" bazel/repositories.bzl | sha1sum | head -c 40)
+        echo "GHA_BAZEL_CACHE_KEY=${GHA_BAZEL_CACHE_KEY}" >> $GITHUB_ENV
```

Comment:
I must admit that this is ugly but I couldn't find the proper solution to this.. I guess there's some bazel query to get proper cache keys for a specific target..

Reply:
I don't think `bazel query` understands `--define`, so that wouldn't be helpful anyway.

But I think that we could simply use:
```
key: bazel-${{ hashFiles('WORKSPACE', '.bazelrc', '.bazelversion', 'bazel/cargo/Cargo.raze.lock', 'bazel/dependencies.bzl', 'bazel/repositories.bzl') }}-${{ matrix.runtime }}
```
since Bazel and other dependencies (e.g. `rules_foreign_cc`) also affect cache, and we're not updating those files that often, so we don't have to worry about invalidating cache too much, IMHO.

Reply:
yeah agreed. Updated.